src/imitation/algorithms/bc.py

```diff
@@ -366,10 +375,23 @@ def train(
                 for stats in [stats_dict_it, stats_dict_loss]:
                     for k, v in stats.items():
                         logger.record(k, v)
+                # TODO(shwang): Maybe instead use a callback that can be shared between
+                # all algorithms' `.train()` for generating rollout stats.
+                if log_rollouts_venv is not None and log_rollouts_n_episodes > 0:
```

Comment:
This does seem like it might be a better solution. Does [EvalCallback](https://stable-baselines3.readthedocs.io/en/master/guide/callbacks.html#evalcallback) do what you need? It does log [episode reward and length](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/common/callbacks.html).

The flip-side is I think every other algorithm we have logs stuff like this without a callback? So perhaps it does make sense this is built into BC.

Reply:
EvalCallback does seem like a good fit -- adding it to the comment here. For better SB3 integration it could be nice to add Callbacks to every imitation alg's `train` someday.